### ISSUE TITLE
Word add-in tutorial. Security certificate error.

### ISSUE THREAD

**Issue by JohnLukeBentley** [OfficeDev/office-js-docs-pr#1275]
## Article URL
["Tutorial: Create a Word task pane add-in" ... "Test the add-in"](https://docs.microsoft.com/en-us/office/dev/add-ins/tutorials/word-tutorial#test-the-add-in)

## Issue

In short: In Word for Desktop I get a security certificate error when trying to the follow the tut.

In detail:

I've successfully completed the [Build your first Word task pane add-in](https://docs.microsoft.com/en-us/office/dev/add-ins/quickstarts/word-quickstart?tabs=yeomangenerator), using:

* "Yeoman generator for Office Add-ins";
* Sideloading on Windows (for used in Word on Windows Desktop); and
* With the project in Visual Studio *Code*.

However, I get an error when attempting ["Tutorial: Create a Word task pane add-in"](https://docs.microsoft.com/en-us/office/dev/add-ins/tutorials/word-tutorial#test-the-add-in). (This tutorial avoids the "Yeoman generator for Office Add-ins", if I'm understanding matters correctly) . Specifically when attempting  to sideload on Windows  I get the following error in my Taskpane:

    ADD-IN ERROR
    The content is blocked because it isn't signed by a valid security certificate.
    [Restart]

In following "Tutorial: Create a Word task pane add-in" I've Installed Word-Add-in-Tutorial\Start\certs\ca.crt by, as directed, following [Installing the self-signed certificate](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md), using "Method #2: manually install the certificate" for Windows. In short, by double clicking ca.crt and installing it into the "Trusted Root Certification Authorities" store.

That is, it seems I've successfully trusted the Certificate Authority. However I haven't signed the project as such (and am unclear if that is required and how one would do that if it is). In \certs\ I see the: ca.crt; server.crt; and server.key. 

Is there a missing step in the documentation that requires one to sign the project (quite apart from installing the ca.crt for Certificate Authority trust)?

**Comment by kbrandl** [comment]
@JohnLukeBentley sorry to hear that you've run into this problem. @Rick-Kirkham is this something that you can investigate? Thanks!

**Comment by JohnLukeBentley** [comment]
Thanks @kbrandl !

**Comment by Rick-Kirkham** [comment]
I can reproduce this. I'm investigating.

**Comment by kbrandl** [comment]
This cause of this issue may be that there are extraneous/old/untrusted certificates for **localhost** in the certificate store on your machine -- as described in [my last comment](https://github.com/OfficeDev/generator-office/issues/486#issuecomment-528565677) on this issue: https://github.com/OfficeDev/generator-office/issues/486. I was able to resolve the issue described there by manually removing the extraneous **localhost** certs -- perhaps give that a try?

**Comment by Rick-Kirkham** [comment]
I've tried that. But it found the old cert anyway. Still working on it.

**Comment by Rick-Kirkham** [comment]
I can reproduce this on one of my computers, but everything works fine on another. We expect to be making a change to the tutorial in the next few days that will change the cert handling significantly anyway. I think we're not likely to get a shorter term fix.

**Comment by JohnLukeBentley** [comment]
@kbrandl. Thanks for that suggestion. I tried a few variations on that suggestions with no luck. Anyway ....

@Rick-Kirkham. Thanks. I'll wait for the cert handling overhaul.

**Comment by Rick-Kirkham** [comment]
@JohnLukeBentley This fixed it on my machine. Could you see if it fixes it for you? 
1. Close all browsers and all Office apps.
2. If the server is running, shut it down by entering Ctrl-C twice. Then close the command prompt window.
3. Open the bconfig.json file in the root of the project.
4. Replace the line `"https": true,` with the following:

```
"https": {
        "key": "./certs/server.key",
        "cert": "./certs/server.crt"
    },
```

5. Repeat the "Test the add-in" section of the tutorial.

**Comment by JohnLukeBentley** [comment]
That fixed it! @Rick-Kirkham, you are a champion!

In addition to acting in conformance with your steps above, I  (re)Installed Start\certs\ca.crt by, as directed in the Tut linked [Installing the self-signed certificate](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md), by double clicking it and installing it to the certificate store in "Trusted Root Certification Authorities". This displays as "Issued To: localhost-ca".

I did this given my fiddling with my certs after @kbrandl's helpful suggestion.

Does your fix count as the "significant" change to cert handling, or is there some other pending overhaul to be done?

**Comment by Rick-Kirkham** [comment]
@kbrandl is working on a more drastic improvement. The tutorial won't be based on cloning a repo. Instead, you'll use our Yo Office tool to create a project. There's a point in that process, where the tool will just trust a cert automatically.

**Comment by Rick-Kirkham** [comment]
Closing this as fixed because the fix goes to the repo used by the tutorial, not this repo for our docs.

**Comment by JohnLukeBentley** [comment]
Yeah, using the Yeoman tool would make the tut consistent with the immediately prior "Build your first Word task pane add-in". So that overhaul of the tut sounds like the right thing to do.

Your fix allows me to keep progressing through the tut in its current state. So I'm grateful for that.

By the way, since last posting I reset my local changes, fetched from the updated repo, and restarted the tut. And it all works fine so far: I can work passed the "Test the add-in" in virtue of the incorporated cert references in `bsconfig.json`.

In short, yes I think it right for you to close this as fixed. 

Thanks for your help!
